resources/views/livewire/auth/two-factor-challenge.blade.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+use App\Actions\TwoFactorAuth\VerifyTwoFactorCode;
+use App\Actions\TwoFactorAuth\ProcessRecoveryCode;
+use App\Actions\TwoFactorAuth\CompleteTwoFactorAuthentication;
+use App\Actions\TwoFactorAuth\GetTwoFactorAuthenticatableUser;
+use Illuminate\Support\Facades\Route;
+use Livewire\Attributes\On;
+use Livewire\Attributes\Validate;
+use Livewire\Attributes\Layout;
+use Livewire\Volt\Component;
+
+new #[Layout('components.layouts.auth')] class extends Component
+{
+    public $recovery = false;
+
+    #[Validate('required|min:6')] 
+    public $auth_code;
+    public $recovery_code;
+
+    public function mount()
+    {
+        if ( !session()->has('login.id')) {
+            return redirect()->route('login');
+        }
+        $this->recovery = false;
+    }
+
+    #[On('submitCode')] 
+    public function submitCode($code)
+    {
+        $this->auth_code = $code;
+        $this->validate();
+
+        // Get the user that is in the process of 2FA
+        $user = app(GetTwoFactorAuthenticatableUser::class)();
+        
+        if (!$user) {
+            return redirect()->route('login');
+        }
+        
+        // Verify the authentication code
+        $valid = app(VerifyTwoFactorCode::class)(decrypt($user->two_factor_secret), $code);
```

Comment:
There is no rate limiting here, which would allow an attacker to brute-force the 2FA code.

Reply:
Added rate limiting in the latest commit. Thanks!